test/metabase/upload_test.clj

```diff
@@ -619,136 +639,139 @@
+                           "b,Millennium Falcon,Han Solo,Blaster"]))
+          (testing "Table and Fields exist after sync"
+            (sync/sync-database! (mt/db))
+            (let [table (t2/select-one Table :db_id (mt/id))]
+              (is (=? {:name #"(?i)upload_test"} table))
+              (testing "Check the data was uploaded into the table correctly"
+                (is (= ["id" "ship" "name" "weapon"]
+                       (column-names-for-table table)))
+                (is (=? {:name                       #"(?i)id"
+                         :semantic_type              :type/PK
+                         :base_type                  :type/Text
+                         :database_is_auto_increment false}
+                        (t2/select-one Field :database_position 0 :table_id (:id table))))))))))))
 
 (deftest load-from-csv-reserved-db-words-test
   (testing "Upload a CSV file with column names that are reserved by the DB"
     (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
-      (mt/with-empty-db
-        (with-mysql-local-infile-activated
+      (with-mysql-local-infile-on-and-off
+        (mt/with-empty-db
           (upload/load-from-csv!
            driver/*driver*
            (mt/id)
            "upload_test"
            (csv-file-with ["id,ship,captain"
                            "1,Serenity,Malcolm Reynolds"
-                           "2,Millennium Falcon, Han Solo"])))
-        (testing "Table and Fields exist after sync"
-          (sync/sync-database! (mt/db))
-          (let [table (t2/select-one Table :db_id (mt/id))]
-            (is (=? {:name #"(?i)upload_test"} table))
-            (testing "Check the data was uploaded into the table correctly"
-              (is (= ["id", "ship", "captain"]
-                     (column-names-for-table table))))))))))
+                           "2,Millennium Falcon, Han Solo"]))
+          (testing "Table and Fields exist after sync"
+            (sync/sync-database! (mt/db))
+            (let [table (t2/select-one Table :db_id (mt/id))]
+              (is (=? {:name #"(?i)upload_test"} table))
+              (testing "Check the data was uploaded into the table correctly"
+                (is (= ["id", "ship", "captain"]
+                       (column-names-for-table table)))))))))))
 
 (deftest load-from-csv-missing-values-test
   (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
-    (mt/with-empty-db
-      (with-mysql-local-infile-activated
+    (with-mysql-local-infile-on-and-off
+      (mt/with-empty-db
         (testing "Can upload a CSV with missing values"
           (upload/load-from-csv!
            driver/*driver*
            (mt/id)
            "upload_test"
-           (csv-file-with ["id,column_that_doesnt_have_a_value" "2"]))
+           (csv-file-with ["id,column_that_has_one_value,column_that_doesnt_have_a_value"
+                           "1,2"
+                           "2,  ,\n"]))
```

Comment:
Apart from the macro changes, this is the only change to the `upload_test`. I wanted to test for empty and blank values too, so I added more rows and columns.